File: bender/app/src/main/java/kz/app/bender/support/stat/Zprefs.java
package kz.app.bender.support.stat;

import android.content.SharedPreferences;
import android.support.annotation.Nullable;

import com.google.gson.Gson;
import com.orhanobut.hawk.Hawk;

import kz.app.bender.models.TSignIn;
import kz.app.bender.support.Config;

class Zprefs {

    public static String getHost(SharedPreferences preference) {
//        if (BuildConfig.DEBUG) {
        //TODO: if project have beta data base
//            return getDebugHost(preference);
//        } else {
        return Config.HOST_BASE;
//        }
    }

    public static void setUser(TSignIn signIn, SharedPreferences preferences) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(Zvars.PREFERENCE_USER_DATA, new Gson().toJson(signIn));
        editor.apply();
    }

    @Nullable
    public static TSignIn getUser(SharedPreferences preferences) {
        String str = preferences.getString(Zvars.PREFERENCE_USER_DATA, "");
        TSignIn userData = null;
        if (!str.isEmpty()) {
            try {
                userData = new Gson().fromJson(str, TSignIn.class);
            } catch (Exception ignored) {

            }
        }
        return userData;
    }

    public static void clearUser(SharedPreferences preferences) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.remove(Zvars.PREFERENCE_USER_DATA);
        editor.apply();
    }

    public static void setUserAuthToken(String token) {
        Hawk.put(Zvars.PREFERENCE_SECURE_AUTH_TOKEN_V_1, token);
    }

    public static String getUserAuthToken() {
        return Hawk.get(Zvars.PREFERENCE_SECURE_AUTH_TOKEN_V_1);
    }

    public static void clearUserAuthToken() {
        Hawk.delete(Zvars.PREFERENCE_SECURE_AUTH_TOKEN_V_1);
    }
}

File: Alem/app/src/main/java/kz/alemtime/alem/support/stat/ZPrefs.kt
package kz.alemtime.alem.support.stat

import android.content.SharedPreferences
import com.google.gson.Gson
import com.orhanobut.hawk.Hawk
import kz.alemtime.alem.models.TProfile
import java.util.*

class ZPrefs {

    companion object {

        //region AUTHORIZATION
        var HOST: String = "https://api.alemtime.kz/"

        fun setUserAuthToken(token: String?) {
            Hawk.put(ZVars.PREFERENCE_SECURE_AUTH_TOKEN_V_1, token)
        }

        fun getUserAuthToken(): String? {
            return Hawk.get(ZVars.PREFERENCE_SECURE_AUTH_TOKEN_V_1)
        }

        fun clearUserAuthToken() {
            Hawk.delete(ZVars.PREFERENCE_SECURE_AUTH_TOKEN_V_1)
        }

        fun setUserAuthTokenTTL(ttl: Date?) {
            Hawk.put(ZVars.PREFERENCE_SECURE_AUTH_TOKEN_TTL_V_1, ttl)
        }

        fun getUserAuthTokenTTL(): Date? {
            return Hawk.get(ZVars.PREFERENCE_SECURE_AUTH_TOKEN_TTL_V_1)
        }
        //endregion

        //region PROFILE
        fun setUserProfile(profile: TProfile, preferences: SharedPreferences) {
            preferences.edit()
                    .putString(ZVars.PREFERENCE_PROFILE, Gson().toJson(profile))
                    .apply()
        }

        fun getUserProfile(sharedPreferences: SharedPreferences): TProfile? {
            val str = sharedPreferences.getString(ZVars.PREFERENCE_PROFILE, "")
            var profile: TProfile? = null
            if (!str!!.isEmpty()) {
                profile = Gson().fromJson<Any>(str, TProfile::class.java) as TProfile?
            }
            return profile
        }

        private fun clearUserProfile(preferences: SharedPreferences) {
            preferences.edit().remove(ZVars.PREFERENCE_PROFILE).apply()
        }

        fun logOut(sharedPreferences: SharedPreferences) {
            clearUserAuthToken()
//            clearUserAuthTokenTTL()
//            clearUserRefreshToken()
            clearUserProfile(sharedPreferences)
        }
        //endregion

    }

}